### PR TITLE
Correct OneCycleLR doc example code to explicitly call optimizer.step()

### LINKED ISSUE #89358: `OneCycleLR` documentation misleading about optimizer
### 📚 The doc issue

The [OneCycleLR documentation](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html) provides example code that doesn't trigger `optimizer.step()`:

```py
data_loader = torch.utils.data.DataLoader(...)
optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
scheduler = torch.optim.lr_scheduler.OneCycleLR(optimizer, max_lr=0.01, steps_per_epoch=len(data_loader), epochs=10)
for epoch in range(10):
    for batch in data_loader:
        train_batch(...)
        scheduler.step()
```

This is confusing, because the `OneCycleLR` scheduler does engulf the optimizer and could therefore plausibly trigger the `optimizer.step()` call for the user. So the omission seems intentional at first glance. But calling `optimizer.step()` [is in fact necessary](https://stackoverflow.com/q/59996859/199475).

### Suggest a potential alternative/fix

The `OneCycleLR` documentation could just state that calling `optimizer.step()` is still needed, or could show the `optimizer.step()` call in the pseudocode.

cc @svekars @carljparker @vincentqb @jbschlosser @albanD

### PR DESCRIPTION
**Pull request by kiersten-stokes**
Fixes #89358 as suggested in the issue comment

A screenshot of the example code in the built docs:
Screenshot 2023-02-28 at 4 46 45 PM

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
This looks great! When the Python docs are finished running, could you paste a picture of what the new docs look like in your PR body to ensure they render properly? A link to the docs can be found in the pytorch-bot comment on the PR under Helpful Links.

**Review by janeyx99 — APPROVED** [review]

**Comment by kiersten-stokes** [comment]
> When the Python docs are finished running, could you paste a picture of what the new docs look like in your PR body to ensure they render properly? A link to the docs can be found in the pytorch-bot comment on the PR under Helpful Links.

@janeyx99 Done! And thank you for the suggestion - I'll keep this in mind for any future doc contributions as well!
